manticore/platforms/linux.py

```diff
@@ -449,25 +461,32 @@ class Linux(Platform):
     BASE_DYN_ADDR_32 = 0x56555000
     BASE_DYN_ADDR = 0x555555554000
 
-    def __init__(self, program, argv=None, envp=None, disasm="capstone", **kwargs):
+    def __init__(
+        self,
+        program: Optional[str],
+        argv: List[str] = [],
+        envp: List[str] = [],
+        disasm: str = "capstone",
+        **kwargs,
+    ):
         """
         Builds a Linux OS platform
         :param string program: The path to ELF binary
         :param string disasm: Disassembler to be used
         :param list argv: The argv array; not including binary.
         :param list envp: The ENV variables.
         :ivar files: List of active file descriptors
-        :type files: list[Socket] or list[File]
+        :type files: list[Socket or File or None]
         """
         super().__init__(path=program, **kwargs)
 
         self.program = program
         self.clocks = 0
-        self.files = []
+        self.files: List[Union[None, File, Socket]] = []
         # A cache for keeping state when reading directories { fd: dent_iter }
-        self._getdents_c = {}
-        self._closed_files = []
-        self.syscall_trace = []
+        self._getdents_c: Dict[int, Any] = {}
+        self._closed_files: List[Union[File, Socket]] = []
+        self.syscall_trace: List = []
```

Comment:
Huh. I thought mypy didn't like a `List` without brackets for the type. Must default to `Any` then. 

Here's at least one line populates `syscall_trace`
https://github.com/trailofbits/manticore/blob/7dfaf9300dba9f76280b5edd69c5e1ade084594c/manticore/platforms/linux.py#L1432

So it should be something like 
```suggestion
        self.syscall_trace: List[Tuple[str, int, bytes]] = []
```

Reply:
Yeah, in mypy, a `List` without any parameter is shorthand for `List[Any]`.  Similarly, `Dict` without any parameters is shorthand for `Dict[Any, Any]`:

> Using a generic class without specifying type parameters assumes Any for each position.

From here: https://docs.python.org/3/library/typing.html

Thanks for the more precise type for `syscall_trace` :)